### PR TITLE
Progress bar dialog

### PR DESCRIPTION
**Pull request by asaikarthikeya**
### **Improved Progress Bar dialog's** 

https://user-images.githubusercontent.com/70392921/122927185-40224c00-d386-11eb-8820-0c449bfcfd8f.mp4

### CODE REVIEW

File: org.envirocar.app/src/org/envirocar/app/views/dashboard/DashboardFragment.java [hunk 1]
```diff
@@ -444,16 +447,31 @@ protected void onStartTrackButtonClicked() {
...
 
+                        // If the device is not found to start the track, dismiss the Dialog in 15 sec
+                        new CountDownTimer(15000, 1000) {
+                            @Override
+                            public void onTick(long millisUntilFinished) {
+                            }
+                            @Override
+                            public void onFinish() {
+                                connectingDialog.dismiss();
+                                getActivity().stopService(obdRecordingIntent);
```
Comment (SebaDro) [inline org.envirocar.app/src/org/envirocar/app/views/dashboard/DashboardFragment.java:468]: You could also use the existing `ServiceUtils` class with `  ServiceUtils.stopService(...)`.

File: org.envirocar.app/src/org/envirocar/app/views/logbook/LogbookAddFuelingFragment.java [hunk 2]
```diff
@@ -412,16 +414,16 @@ private void uploadCarBeforeFueling(final Car car, final Fueling fueling) {
...
 
                     @Override
                     protected void onStart() {
                         LOG.info("uploadCarBeforeFueling() has started");
-                        dialog = new MaterialDialog.Builder(getContext())
-                                .progress(true, 0)
-                                .title(R.string.logbook_dialog_uploading_fueling_header)
-                                .content(R.string.logbook_dialog_uploading_fueling_car)
-                                .cancelable(false)
+                        dialog = DialogUtils.createDefaultDialogBuilder(getContext(),
```
Comment (SebaDro) [inline org.envirocar.app/src/org/envirocar/app/views/logbook/LogbookAddFuelingFragment.java:422]: Same here: do not use a deprecated method.

File: org.envirocar.app/src/org/envirocar/app/views/logbook/LogbookAddFuelingFragment.java [hunk 3]
```diff
@@ -471,16 +473,16 @@ private void uploadFueling(Fueling fueling) {
...
 
                     @Override
                     public void onStart() {
                         LOG.info("Started the creation of a fueling.");
-                        dialog = new MaterialDialog.Builder(getContext())
-                                .progress(true, 0)
-                                .title(R.string.logbook_dialog_uploading_fueling_header)
-                                .content(R.string.logbook_dialog_uploading_fueling_content)
-                                .cancelable(false)
+                        dialog = DialogUtils.createDefaultDialogBuilder(getContext(),
```
Comment (SebaDro) [inline org.envirocar.app/src/org/envirocar/app/views/logbook/LogbookAddFuelingFragment.java:481]: And here...

File: org.envirocar.app/src/org/envirocar/app/views/login/SigninActivity.java [hunk 4]
```diff
@@ -179,14 +181,15 @@ private void login(String username, String password) {
...
-                    private MaterialDialog dialog;
+                    private AlertDialog dialog;
 
                     @Override
                     protected void onStart() {
-                        dialog = new MaterialDialog.Builder(SigninActivity.this)
-                                .title(R.string.activity_login_logging_in_dialog_title)
-                                .progress(true, 0)
-                                .cancelable(false)
+                        dialog = DialogUtils.createDefaultDialogBuilder(SigninActivity.this,
```
Comment (SebaDro) [inline org.envirocar.app/src/org/envirocar/app/views/login/SigninActivity.java:188]: Same here...

**Review by SebaDro — CHANGES_REQUESTED** [review]

File: org.envirocar.app/src/org/envirocar/app/views/dashboard/DashboardFragment.java (hunk 1 above)
Comment (asaikarthikeya) [reply org.envirocar.app/src/org/envirocar/app/views/dashboard/DashboardFragment.java:468]: Initially, i also used `ServiceUtils.stopService()` to stop the recording but got some error's in `ServiceUtils `class. 

Screenshot from 2021-06-23 15-50-57

Comment (SebaDro) [reply org.envirocar.app/src/org/envirocar/app/views/dashboard/DashboardFragment.java:468]: Have a look on how it's done with `startService()` in the `ServiceUtils` class. Maybe, it's helpful to overload `StopService` in a similar manner.

**Review by SebaDro — CHANGES_REQUESTED** [review]

File: org.envirocar.core/src/main/java/org/envirocar/core/utils/ServiceUtils.java [hunk 5]
```diff
@@ -52,15 +52,21 @@ public static final void startService(Context context, Intent serviceIntent) {
     }
 
     public static final void stopService(Context context, Class<?> service) {
-        LOGGER.info("Trying to stop %s", service.getSimpleName());
+        Intent serviceIntentstop = new Intent(context, service);
+        startService(context, serviceIntentstop);
```
Comment (SebaDro) [inline org.envirocar.core/src/main/java/org/envirocar/core/utils/ServiceUtils.java]: Seems like a copy&paste mistake. Please, be a little more attentive when changing the code in order to reduce the number of review cycles.

**Review by SebaDro — CHANGES_REQUESTED** [review]

File: org.envirocar.core/src/main/java/org/envirocar/core/utils/ServiceUtils.java (hunk 5 above)
Comment (asaikarthikeya) [reply org.envirocar.core/src/main/java/org/envirocar/core/utils/ServiceUtils.java]: Updated

**Comment by asaikarthikeya** [comment]
https://user-images.githubusercontent.com/70392921/123134270-3e33b800-d46e-11eb-97ff-48c5176e6fbc.mp4

**Review by SebaDro — APPROVED** [review]

**Pull request merged**